src/wms/src/main/java/org/geoserver/wms/kvp/MapLayerInfoKvpParser.java

```diff
@@ -33,6 +33,13 @@ public MapLayerInfoKvpParser(final String key, final WMS wms) {
         rawNamesParser = new FlatKvpParser(key, String.class);
     }
 
+    /**
+     * Returns whether the specified resource is valid in the context of the current request.
+     */
+    protected boolean passRequestedObject(Object theResource) {
```

Comment:
By your description it seems a "skipResource" name would be better, but I'll leave this to native English speakers to confirm.